sycl/include/CL/sycl/detail/accessor_impl.hpp

```diff
@@ -115,8 +115,8 @@ class AccessorBaseHost {
   const range<3> &getMemoryRange() const { return impl->MMemoryRange; }
   void *getPtr() const { return const_cast<void *>(impl->MData); }
 
-  template <class Obj>
-  friend decltype(Obj::impl) detail::getSyclObjImpl(const Obj &SyclObject);
+  template <class T>
+  friend decltype(T::impl) getSyclObjImpl(const T &SyclObject);
```

Comment:
This is an interesting test case.

MSVC 2019 did not recognize getSyclObjImpl as a function. Removing 'detail::' seems reasonable here as the class AccessorBaseHost is already in cl::sycl::detail space. It helped to recognize it as a func.

Right after that something really odd happened. MSVC started complaining about having 2 funcs (this declaration at line 119, and the real function definition in common.hpp), and inability to resolve an ambiguous call. 
Simple replacement of the template parameter NAME from 'Obj' to 'T' in this declaration (i.e. making this friend func declaration looking literally identical to the func definition in common.hpp) eliminated the error.
Seriously? Why? Probably it is just a bug in MSVC2019. 
If someone has different opinion why such things may work incorrectly please let me know to void similar problems in future.